### PR TITLE
Protect run config uploads with typed summaries

### PR DESCRIPTION
**Pull request by kcoopermiller**
## What this changes

prime-runs no longer uploads run configs verbatim. A launch file's comments, credentials, internal URLs, harness environment values or private task data could otherwise be stored as run metadata. The SDK keeps the original in the process (`run.config` / `run.config_source`) and sends only a typed summary.

## What reaches the dashboard

**Shown when present and correctly typed:**
- Run controls (numbers): `num_examples`, `rollouts_per_example`, `num_tasks`, `num_rollouts`, `max_concurrent`, `max_retries`, `timeout_minutes`, `max_tokens`, `temperature`, `top_k`, `top_p`, `frequency_penalty`, `presence_penalty`, `repetition_penalty`, `seed`
- Flags (booleans): `auto_max_concurrent`, `independent_scoring`, `verbose`, `shuffle`, `push`, `allow_sandbox_access`, `allow_instances_access`, `allow_tunnel_access`, `use_platform_billing`
- `sampling` / `sampling_args`: `max_tokens`, `temperature`, `top_k`, `top_p`, `frequency_penalty`, `presence_penalty`, `repetition_penalty`, `seed`, `min_p`, `enable_thinking` (bool), `reasoning_effort` (`none`/`minimal`/`low`/`medium`/`high`/`xhigh`)
- `client.type` (`eval` or `train`); `serve.max_concurrent` and `serve.pool` (`type`, `num_workers`, `max_workers`, `multiplex`)
- Metadata: `avg_reward`, `avg_score`; `state_columns` names only (first 128 entries, each up to 128 characters); `prime_runs.status` and `prime_runs.finished_at` (must carry a timezone, shown in UTC)
- External training registration: the same controls plus `max_steps`, `batch_size`, `seq_len`, `lr`, `learning_rate`, `weight_decay`, `warmup_steps` (top level and under `trainer`)

## What never leaves the process

- Launch file text and filenames, URLs, paths, commands
- Task, harness and environment data, headers, harness `env` values
- Any key not listed above, including custom strings
- Values with the wrong type under a known key (a string under `max_tokens`, an object under `seed`): dropped, not coerced

A run that fails or is cancelled is closed through the status API with the message passed to `run.fail`; no config values travel with it.

---

> [!NOTE]
> **Medium Risk**
> Changes what evaluation and training runs persist remotely and may leave dashboards with less config detail until callers pass summary mappings; mitigated by allowlists and broad test coverage of wire bodies.
> 
> **Overview**
> **Run config is no longer sent verbatim to the platform.** Eval create/update/failure paths and external training registration now pass **`metadata_summary` / `training_config_summary`** filters so only allowlisted numeric/boolean controls, sampling settings, client type, serve/pool fields, bounded `state_columns` names, and sanitized terminal `prime_runs` timestamps reach the API. Launch file text, `config_source`, URLs, paths, harness env, headers, and other unknown keys stay on the client in `run.config` / `run.config_source`.
> 
> This is a **breaking contract change** for dashboard metadata: a file-only `config=` path uploads no remote controls unless callers merge summary fields in a mapping (README example updated). **`EvalsBackend.update`** always sends `metadata` (including `{}`) when config is provided so stale full configs can be cleared.
> 
> New **`config_privacy`** module plus fixture-driven tests assert request bodies never contain secret-like payloads and that training `trainer.lr` still projects safely.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e8a79451a2d3d286a66d5763cbc9f20b0776eb23. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/prime-runs/src/prime_runs/config_privacy.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    result = config_summary(value)
+    if not isinstance(value, dict):
+        return result
+    result.update(_numbers(value, ("avg_reward", "avg_score")))
+    columns = value.get("state_columns")
+    if isinstance(columns, list):
+        result["state_columns"] = [
+            column
+            for column in columns[:128]
+            if isinstance(column, str) and re.fullmatch(r"[A-Za-z_][A-Za-z0-9_]{0,127}", column)
```
[the remark this answers is not in the document]
Comment (kcoopermiller) [reply packages/prime-runs/src/prime_runs/config_privacy.py]: Valid; fixed in e19028e2, with matching platform API/frontend changes in platform #5238 (28fa0b8623). The projection preserves exact non-empty sample column names, including hyphens, dots, spaces and Unicode, within the existing 128-name / 128-code-point bounds. Invalid entries are omitted independently. Shared fixtures exercise tool-output, grader.score, ground truth and Unicode through SDK create/update requests and API/browser projections; boundary tests cover empty/oversized names and list limits. All 277 prime-runs tests, 50 focused backend tests and 10 frontend tests pass. Column values and raw config sources are still excluded.

File: packages/prime-runs/src/prime_runs/config_privacy.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    """Copy only supported, correctly typed controls. Never parse raw files."""
+    if not isinstance(value, dict):
+        return {}
+    result = _numbers(value, NUMBER_FIELDS)
+    result.update({key: value[key] for key in BOOL_FIELDS if type(value.get(key)) is bool})
+    for key in ("sampling", "sampling_args"):
+        sampling = value.get(key)
+        if isinstance(sampling, dict):
+            projected = _numbers(sampling, SAMPLING_FIELDS)
+            effort = sampling.get("reasoning_effort")
```
[the remark this answers is not in the document]
Comment (kcoopermiller) [reply packages/prime-runs/src/prime_runs/config_privacy.py:78]: Fixed in e8a79451: `sampling.enable_thinking` / `sampling_args.enable_thinking` is copied when it is exactly a `bool` (tests cover both keys, both values, and non-boolean rejects). The same field is added to the platform projection on #5238 so the API keeps it too.

**Comment by kcoopermiller** [comment]
@codex review
